plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/MavenPath.java

```diff
@@ -335,6 +361,69 @@ public MavenPath signature(final SignatureType signatureType) {
     );
   }
 
+  /**
+   * Returns path pointing to given extension and optionally classifier within this same GAV. Only usable for artifact
+   * paths, those having non-null {@link #getCoordinates()}.
+   */
+  @Nonnull
+  public MavenPath locate(final String extension, @Nullable final String classifier) {
+    checkNotNull(extension);
+    checkArgument(coordinates != null, "Only artifact paths may locate: %s", this);
+
+    MavenPath origin = this;
+    if (isHash()) {
+      origin = origin.subordinateOf();
+    }
+    if (isSignature()) {
+      origin = origin.subordinateOf();
+    }
+    Coordinates coordinates = new Coordinates(
+        origin.coordinates.isSnapshot(),
+        origin.coordinates.getGroupId(),
+        origin.coordinates.getArtifactId(),
+        origin.coordinates.getVersion(),
+        origin.coordinates.getTimestamp(),
+        origin.coordinates.getBuildNumber(),
+        origin.coordinates.getBaseVersion(),
+        classifier,
+        extension,
+        null
```

Comment:
@kellyrob99 re "seems fishy": the diff above is not true, seems a GH bug, see here for "real" file:
https://github.com/sonatype/nexus-oss/pull/1257/files#r29603557

The signature type is nullified by call to `main()`, which basically moves "up" to a non-subordinate path (is not a hash nor signature), and then "locates" the given ext and classifier, Maybe the javadoc needs update, but this method will never return subordinate path. Example:

```java
  MavenPath path = ... // /org/corp/coolstuff/1.0/coolstuff-1.0.pom.asc.sha1
  // get the sources path whatever path above points to
  path = path.locate("jar", "sources"); // result: /org/corp/coolstuff/1.0/coolstuff-1.0-sources.jar
  // get the sources hash
  path = path.hash(HashType.SHA1); // result: /org/corp/coolstuff/1.0/coolstuff-1.0-sources.jar.sha1
  
```